go/test/endtoend/vreplication/time_zone_test.go

```diff
@@ -0,0 +1,194 @@
+	vc.AddKeyspace(t, []*Cell{cell1}, sourceKs, "0", initialProductVSchema, initialProductSchema, 0, 0, 100, sourceKsOpts)
+
+	vtgate = cell1.Vtgates[0]
+	require.NotNil(t, vtgate)
+	vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", "product", "0"), 1)
+
+	vtgateConn = getConnection(t, vc.ClusterConfig.hostname, vc.ClusterConfig.vtgateMySQLPort)
+	defer vtgateConn.Close()
+	verifyClusterHealth(t, vc)
+
+	productTab := vc.Cells[defaultCell.Name].Keyspaces[sourceKs].Shards["0"].Tablets["zone1-100"].Vttablet
+	timeZoneSQLBytes, _ := os.ReadFile("tz.sql")
+	timeZoneSQL := string(timeZoneSQLBytes)
+
+	// it seems to take some time for the mysql server to load time zone info after the tables in mysql db have been populated
+	loadTimeZoneInfo := func(tab *cluster.VttabletProcess, sql, timezone string) {
+		_, err := tab.QueryTabletWithDB(timeZoneSQL, "mysql")
+		require.NoError(t, err)
+		timer := time.NewTimer(5 * time.Second)
+		for {
+			select {
+			case <-timer.C:
+				require.Fail(t, "could not load time zone info")
+			default:
+			}
+			_, err = tab.QueryTablet(fmt.Sprintf("SET GLOBAL time_zone = '%s';", timezone), "", false)
+			if err == nil {
+				timer.Stop()
+				return
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
+	loadTimeZoneInfo(productTab, timeZoneSQL, "US/Pacific")
+
+	insertInitialData(t)
+
+	if _, err := vc.AddKeyspace(t, cells, targetKs, "0", customerVSchema, customerSchema, defaultReplicas, defaultRdonly, 200, targetKsOpts); err != nil {
+		t.Fatal(err)
+	}
+	if err := vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.primary", "customer", "0"), 1); err != nil {
+		t.Fatal(err)
+	}
+
+	defaultCell := vc.Cells["zone1"]
+	custKs := vc.Cells[defaultCell.Name].Keyspaces[targetKs]
+	customerTab := custKs.Shards["0"].Tablets["zone1-200"].Vttablet
+
+	loadTimeZoneInfo(customerTab, timeZoneSQL, "UTC")
+
+	tables := "datze"
+
+	output, err := vc.VtctlClient.ExecuteCommandWithOutput("MoveTables", "--", "--source", sourceKs, "--tables",
+		tables, "--source_time_zone", "US/Pacific", "Create", ksWorkflow)
+	require.NoError(t, err, output)
+
+	catchup(t, customerTab, workflow, "MoveTables")
+
+	// inserts to test date conversions in replication (vplayer) mode
+	_, err = vtgateConn.ExecuteFetch("insert into datze(id, dt2) values (11, '2022-01-01 10:20:30')", 1, false) // standard time
+	require.NoError(t, err)
+	_, err = vtgateConn.ExecuteFetch("insert into datze(id, dt2) values (12, '2022-04-01 5:06:07')", 1, false) // dst
+	require.NoError(t, err)
+
+	vdiff(t, ksWorkflow, "")
+
+	query := "select * from datze"
+	qrSourceUSPacific, err := productTab.QueryTablet(query, sourceKs, true)
+	require.NoError(t, err)
+	require.NotNil(t, qrSourceUSPacific)
+
+	qrTargetUTC, err := customerTab.QueryTablet(query, targetKs, true)
+	require.NoError(t, err)
+	require.NotNil(t, qrTargetUTC)
+
+	require.Equal(t, len(qrSourceUSPacific.Rows), len(qrTargetUTC.Rows))
+
+	pst, err := time.LoadLocation("US/Pacific")
```

Comment:
Nit, but isn't PST Pacific Standard Time vs. PDT or Pacific Daylight Time? It makes it confusing for me when we later check to see if `pst` is PST or PDT. :-) 

https://github.com/vitessio/vitess/pull/10102/files#diff-b44124544b9ac2d695bf061044407b2f57699f9a63e7ecc2769f6d5a5aac1260R150-R153

Reply:
Changed variable name to `pacificLocation`